invenio_communities/views/api.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Invenio module that adds API to communities."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, abort
+from invenio_rest import ContentNegotiatedMethodView
+
+from invenio_communities.models import Community
+from invenio_communities.serializer import json_serializer
+
+
+blueprint = Blueprint(
+    'invenio_communities_api',
+    __name__,
+    url_prefix='/communities',
+    template_folder='templates',
+    static_folder='static',
+)
+
+
+class CommunitiesResource(ContentNegotiatedMethodView):
+    """"Communities resource."""
+
+    def __init__(self, serializers=None, *args, **kwargs):
+        """Constructor."""
+        super(CommunitiesResource, self).__init__(
+            serializers,
+            *args,
+            **kwargs
+        )
+
+    def get(self):
+        """Get a list of all the communities.
+
+        .. http:get:: /communities/(string:id)
+            Returns a JSON list with all the communities.
+            **Request**:
+            .. sourcecode:: http
+                GET /communities HTTP/1.1
+                Accept: application/json
+                Content-Type: application/json
+                Host: localhost:5000
+            :reqheader Content-Type: application/json
+            **Response**:
+            .. sourcecode:: http
+                HTTP/1.0 200 OK
+                Content-Length: 334
+                Content-Type: application/json
+                [
+                    {
+                        "id": "comm1"
+                    },
+                    {
+                        "id": "comm2"
+                    }
+                ]
+            :resheader Content-Type: application/json
+            :statuscode 200: no error
+        """
+        communities = Community.query.all()
+        result = []
+        for community in communities:
+            result.append({
+                'id': community.id
+            })
+        return {'json': result}
```

Comment:
I think this is likely more an issue in``ContentNegotiatedMethodView``, but it would be better to return just ``result``. You might need to call ``self.make_response`` explicitly here.